tools/codegen/dest/lazy_ir.py

```diff
@@ -0,0 +1,387 @@
+        ir::MakeNode<ir::ops::{ir_node_name(f.func)}>({node_ctor_input_str}, _out_dtype, _out_shape),
+
+        // (whc): experiment on dtype
+        // try always overriding output dtype to match the one ATen says our op should produce.
+        // this diverges from most of the handwritten methods, which often do not override and 
+        // rely on other behavior in the lowering or copy process to make this correct.
+        // (1) evaluate design goal: to always pick the IR's dtype in one place (here)
+        // (2) rationalize this with Google's design, it may be a problem
+        // (3) evaluate perf impact: make sure we're not actually doing casts becuase of this override
+        _out_dtype));
+}};
+"""
+
+
+def compute_lazy_native_function_definition(
+        g: Union[NativeFunctionsGroup, NativeFunction],
+        backend_index: BackendIndex,
+        codegen: List[OperatorName],
+        class_method_name: str,
+) -> List[str]:
+
+    metadata = backend_index.get_kernel(g)
+    if isinstance(g, NativeFunctionsGroup):
+        if metadata is not None and metadata.structured:
+            raise AssertionError("Structured lazy functions are not implemented yet.")
+        else:
+            return list(mapMaybe(lambda f: gen_unstructured_lazy_definition(f, backend_index, codegen, class_method_name), g.functions()))
+    else:
+        x = gen_unstructured_lazy_definition(g, backend_index, codegen, class_method_name)
+        return [] if x is None else [x]
+
+
+def gen_lazy_shape_dtype_decl(f: NativeFunction, backend_index: BackendIndex, codegen: List[OperatorName]) -> Optional[str]:
+    sig = kernel_signature(f, backend_index)
+    metadata = backend_index.get_kernel(f)
+    if f.func.name not in codegen:
+        return None
+    if metadata is None:
+        return None
+    if "legacy::" in metadata.kernel:
+        return None
+
+    # Lazy IR stuff
+    schema = update_schema_for_lazy_ir(f.func)
+    all_types, value_types, scalar_types = separate_value_scalar_types(schema)
+    lazy_tensor_decls_str = lazy_tensor_decls(value_types)
+    node_ctor_input_str = node_ctor_inputs(value_types, scalar_types)
+
+    # Only generate shape/dtype fn for non-structured kernels,
+    # since we just use the meta function for structured kernels
+    if not f.structured and f.structured_delegate is None:
+        shapeFn = DispatcherSignature.from_schema(FunctionSchema(f.func.name,
+                                                                 f.func.arguments,
+                                                                 returns=(Return(name=None,
+                                                                                 type=intArrayT,
+                                                                                 annotation=None),)),
+                                                  prefix='compute_shape_')
+        dtypeFn = DispatcherSignature.from_schema(FunctionSchema(f.func.name,
```

Comment:
yeah, referring to your other comment (https://github.com/pytorch/pytorch/pull/64104#discussion_r708741985), I think the lower version is better than shapeFn/dtypeFn here.

A more heavy-weight suggestion would be to add a new `LazySignature` type in `types.py`, similar to `DispatcherSignature`, that knows how to print it's normal/shape/dtype signatures. I'm not sure if that's worth it though.